src/components/ListItem/ListItem.vue

```diff
@@ -0,0 +1,457 @@
+<!--
+  - @copyright Copyright (c) 2021 Marco Ambrosini <marcoambrosini@pm.me>
+  -
+  - @author Marco Ambrosini <marcoambrosini@pm.me>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+<docs>
+
+# Usage
+```
+<listItem
+		:title="'Title of the element'"
+		:bold="false"
+		:details="'One hour ago'"
+		:counter-number="44"
+		:counter-highlighted="true">
+		<template #icon>
+			<avatar size="44" user="janedoe" display-name="Jane Doe" />
+		</template>
+		<template v-slot:subtitle>
+			In this slot you can put both text and other components such as icons
+		</template>
+		<AppNavigationCounter
+			#counter">
+			7
+		</AppNavigationCounter>
+		<template #actions>
+			<ActionButton>
+				Button one
+			</ActionButton>
+			<ActionButton>
+				Button two
+			</ActionButton>
+			<ActionButton>
+				Button three
+			</ActionButton>
+		</template>
+	</listItem>
+```
+</docs>
+
+<template>
+	<nav-element
+		class="list-item_wrapper"
+		v-bind="navElement">
+		<a
+			:id="anchorId"
+			ref="list-item"
+			:class="{ 'active' : active }"
+			href="#"
+			class="list-item"
+			:aria-label="linkAriaLabel"
+			@mouseover="handleHover"
+			@focus="handleFocus"
+			@blur="handleBlur"
+			@mouseleave="handleMouseleave"
+			@keydown.tab="handleTab"
+			@click="onClick"
+			@keydown.esc="displayActions = false">
+			<!-- @slot This slot is used for the avatar or icon -->
+			<slot name="icon" />
+			<div class="list-item-content">
```

Comment:
```suggestion
			<slot name="icon" />

			<div class="list-item-content">
```

You can space things a lot more :)
Should help making the template more clear

As well as using <!-- --> Comments to describe what those BEM wrappers are or does :wink: